lib/api/src/js/module.rs

```diff
@@ -558,14 +636,33 @@ impl Module {
         ExportsIterator::new(iter, exports.length() as usize)
     }
 
+    /// Returns true if the module is still ok - this will be
+    /// false if the module was passed between threads in a
+    /// way that it became undefined (JS does not share objects
+    /// between threads except via a post_message())
+    pub fn is_ok(&self) -> bool {
+        let val = JsValue::from(&self.module);
+        !val.is_undefined() && !val.is_null()
+    }
```

Comment:
Rather than constructing the Module, shouldn't we just do this check whenever we cross the boundaries and before the Rust `Module` is constructed?

Reply:
@john-sharratt I agree that this `is_ok()` is quite awakard. 

Can we implement that a bit cleaner?

Reply:
this is no longer used and can be removed - we now share modules between threads properly which reduces memory usage significantly

Reply:
Methods removed.